clients/fides-js/src/components/tcf/TcfOverlay.tsx

```diff
@@ -182,9 +226,30 @@ const TcfOverlay: FunctionComponent<OverlayProps> = ({
     [draftIds]
   );
 
+  const privacyNotices = useMemo(
+    () => experience.privacy_notices ?? [],
+    [experience.privacy_notices]
+  );
+
+  const isAllNoticeOnly = privacyNotices.every(
+    (n) => n.consent_mechanism === ConsentMechanism.NOTICE_ONLY
+  );
+
+  const { servedNotices } = useConsentServed({
+    notices: privacyNotices,
+    options,
+    userGeography: fidesRegionString,
+    acknowledgeMode: isAllNoticeOnly,
+    privacyExperience: experience,
+  });
+
   const handleUpdateAllPreferences = useCallback(
-    (enabledIds: EnabledIds) => {
-      const tcf = createTcfSavePayload({ experience, enabledIds });
+    (enabledIds: EnabledIds, servedNotices: LastServedNoticeSchema[]) => {
```

Comment:
I don't think `servedNotices` needs to be an arg here—I think you can just reference it above (since it's function inside the component, it has access to its vars)

Reply:
That's what I tried originally but when I logged out the value of `servedNotices` inside `handleUpdateAllPreferences` it was empty. The only way I could get it to work is by passing it in as a param. Could it be something to do with `useConsentServed`?

Reply:
Never mind, this is working without the `servedNotices` arg